enterprise/server/raft/sender/sender.go

```diff
@@ -114,21 +122,33 @@ func (s *Sender) fetchRangeDescriptorFromMetaRange(ctx context.Context, key []by
 		rangeDescriptor = rd
 		return nil
 	}
-	for retrier.Next() {
-		metaRangeDescriptor := s.GetMetaRangeDescriptor()
-		if metaRangeDescriptor == nil {
-			log.CtxWarningf(ctx, "RangeCache did not have meta range yet (key %q)", key)
-			continue
-		}
-		_, err := s.tryReplicas(ctx, metaRangeDescriptor, fn, rfpb.Header_LINEARIZABLE)
-		if err == nil {
-			return rangeDescriptor, nil
+	if err := s.runOnMetaRange(ctx, fn); err != nil {
+		return nil, err
+	}
+	return rangeDescriptor, nil
+}
+
+func (s *Sender) LookupRangeDescriptorsForPartition(ctx context.Context, partition disk.Partition) ([]*rfpb.RangeDescriptor, error) {
+	res := make([]*rfpb.RangeDescriptor, 0, partition.NumRanges)
+	fn := func(ctx context.Context, c rfspb.ApiClient, h *rfpb.Header) error {
+		partitionPrefix := []byte(filestore.PartitionDirectoryPrefix + partition.ID + "/")
+		req := &rfpb.ScanRequest{
+			Start:    keys.RangeMetaKey(partitionPrefix),
+			End:      keys.RangeMetaKey(keys.MakeKey(partitionPrefix, keys.MaxByte)),
+			ScanType: rfpb.ScanRequest_SEEKGT_SCAN_TYPE,
+			Limit:    int64(partition.NumRanges),
```

Comment:
this seems error prone -- is it possible that we miss stuff this way?

Reply:
I changed the function signature a bit. I'm now passing partition and the limit explicitly. The caller can decide how to use it.